queries.sql

```diff
@@ -23,3 +23,61 @@ WHERE name != 'Gabumon';
 
 SELECT * FROM animals
 WHERE weight_kg >= 10.4 and weight_kg <= 17.3;
+
+
+BEGIN;
+    UPDATE animals SET species = 'unspecified';
+    SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+    UPDATE animals 
+    SET species = 'digimon' WHERE name LIKE '%mon';
+    UPDATE animals 
+    SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+    DELETE FROM animals;
+    SELECT * FROM animals;
+    ROLLBACK;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+    DELETE FROM animals
+    WHERE date_of_birth > '2022-01-01';
+    SAVEPOINT FIRSTPOINT;
+
+    UPDATE animals 
+    SET weight_kg = weight_kg * (-1);
+    ROLLBACK TO FIRSTPOINT;
+    UPDATE animals 
+    SET weight_kg = weight_kg * (-1)
+    WHERE weight_kg < 0;
+COMMIT;
+
+-- number of animals
+SELECT COUNT(*) FROM animals;
+
+-- animals that have never tried to escape
+SELECT COUNT(*) FROM animals
```

Comment:
- [ ] Here, you are trying to get the animals that have never tried to escape. Instead of using `escape_attempts > 0` , it should be `escape_attempts = 0`. Greater than zero is implying the animals that have actually escaped.